# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/layers/wr/WebRenderMessageUtils.h
# Commit: 11451cccc88c
# Full Hash: 11451cccc88c077e8fe763d4231bbf1af5226061
# Author: Andrew Osmond <aosmond@mozilla.com>
# Date: 2023-12-12 04:37:49
# Regressor Bug: 1869346
# File Overlap Count: 5
# Description:
#   Bug 1869346 - Rework how canvas accesses shared surfaces in the compositor process. r=gfx-reviewers,lsalzman
#   
#   This patch adds plumbing to allow sharing ExternalImageId via
#   SurfaceDescriptor. This will be used in a future patch by WebGL and
#   WebGPU to avoid extra copies. It also refactors how CanvasTranslator
# ==============================================================================

diff -r 8043d7693dba -r 11451cccc88c gfx/layers/wr/WebRenderMessageUtils.h
--- a/gfx/layers/wr/WebRenderMessageUtils.h	Mon Dec 11 18:15:34 2023 +0000
+++ b/gfx/layers/wr/WebRenderMessageUtils.h	Mon Dec 11 18:22:21 2023 +0000
@@ -218,6 +218,13 @@
 struct ParamTraits<mozilla::wr::RenderReasons>
     : public PlainOldDataSerializer<mozilla::wr::RenderReasons> {};
 
+template <>
+struct ParamTraits<mozilla::wr::ExternalImageSource>
+    : public ContiguousEnumSerializer<mozilla::wr::ExternalImageSource,
+                                      mozilla::wr::ExternalImageSource::Unknown,
+                                      mozilla::wr::ExternalImageSource::Last> {
+};
+
 }  // namespace IPC
 
 #endif  // GFX_WEBRENDERMESSAGEUTILS_H